### ISSUE TITLE
[Enhancement]: Show warnings when not following best practices

### ISSUE THREAD

**Issue by svenklemm** [timescale/timescaledb#4335]
### What type of enhancement is this?

User experience

### What subsystems and features will be improved?

Query planner

### What does the enhancement do?

Many users are not aware of postgres best practices and create hypertables with suboptimal schemas leading to performance problems later on because inefficient plans are generated.

Warnings could be generated for the following:
- timestamp without timezone usage
- char/varchar usage
- numeric

For reference:
https://www.timescale.com/forum/t/postgresql-schema-design-best-practices/82

**Comment by vanderhoop** [comment]
@sb230132 @svenklemm So, I just came across these warnings when migrating data from a standard table to a hypertable, and it's unclear to me why TEXT would be preferred to VARCHAR, and I've always defaulted to varchar, particularly when I am certain of the size of the incoming data.

I've searched the docs and googled around for the last half hour, and I haven't been able to find anything yet.

These warnings could be improved if they either **a)** explained why these are best practices for a hypertable, or **b)** pointed developers at some documentation with an explanation as to the best practice, because I'm hesitant to make the change unless I understand why, particularly when it goes against what I've always heard about standard relational tables and storage.

Thoughts?
